### PR TITLE
Sanitize prompt and description text

### PR DESCRIPTION
**Pull request by mrfootoyou**
## Summary

This pull request introduces new utilities for normalizing and sanitizing prompt and description text, and applies them throughout the codebase to improve prompt quality and reduce token usage. The main changes include adding `scrub_description` and `strip_comments_from_prompt` helpers, integrating them into prompt construction and rendering, and adding unit tests for these new utilities.

**Prompt and Description Normalization:**

* Added `scrub_description` and `strip_html` utilities in `utils.prompt` to clean up podcast and episode descriptions by removing HTML, timestamps, URLs, and excessive whitespace, and truncating to a safe length. Updated all places where descriptions are passed to LLM prompts to use these helpers. [[1]](diffhunk://#diff-91bd7be2ba4d7e2af06e9e13f41e47bdaba1a6a435aacc09787b41d4b15705d7R70-R132) [[2]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L42-L61) [[3]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L70-R52) [[4]](diffhunk://#diff-adb37bfcd84f53fefa0abed947c6c2e62a80d896f97db6bacbc8ebfc47d9934bL180-R181) [[5]](diffhunk://#diff-ccf1c409636042d80af13e2c59cc357e226a45f59f9cd1697483d27b10bf0373R1604-R1612) [[6]](diffhunk://#diff-ccf1c409636042d80af13e2c59cc357e226a45f59f9cd1697483d27b10bf0373R3029-R3033) [[7]](diffhunk://#diff-dedec7e091d758d19b2ccaf9af1f9cd6fc4e57f95e47b3ace14fa02a176f87b5L1542-R1546)

* Replaced a local HTML-stripping function in `chapters_generator.py` with the new shared `strip_html` utility. [[1]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L42-L61) [[2]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L70-R52)

**Prompt Comment Stripping:**

* Added `strip_comments_from_prompt` to remove HTML-style comments from prompts, supporting both inline and multi-line comments. Integrated this utility wherever prompts are loaded or rendered, including system prompts, review prompts, and user overrides, to ensure comments are not sent to the LLM. [[1]](diffhunk://#diff-91bd7be2ba4d7e2af06e9e13f41e47bdaba1a6a435aacc09787b41d4b15705d7R70-R132) [[2]](diffhunk://#diff-ccf1c409636042d80af13e2c59cc357e226a45f59f9cd1697483d27b10bf0373L767-R770) [[3]](diffhunk://#diff-ccf1c409636042d80af13e2c59cc357e226a45f59f9cd1697483d27b10bf0373R786) [[4]](diffhunk://#diff-ccf1c409636042d80af13e2c59cc357e226a45f59f9cd1697483d27b10bf0373R801) [[5]](diffhunk://#diff-ae806ae49647b206b91c1031bd3571e9f5fe702f741cb90f3408897579ae12b4L12-R14) [[6]](diffhunk://#diff-ae806ae49647b206b91c1031bd3571e9f5fe702f741cb90f3408897579ae12b4L166-R168) [[7]](diffhunk://#diff-ae806ae49647b206b91c1031bd3571e9f5fe702f741cb90f3408897579ae12b4L212-R214) [[8]](diffhunk://#diff-dedec7e091d758d19b2ccaf9af1f9cd6fc4e57f95e47b3ace14fa02a176f87b5L37-R40) [[9]](diffhunk://#diff-dedec7e091d758d19b2ccaf9af1f9cd6fc4e57f95e47b3ace14fa02a176f87b5R1606) [[10]](diffhunk://#diff-dedec7e091d758d19b2ccaf9af1f9cd6fc4e57f95e47b3ace14fa02a176f87b5R1622-R1629) [[11]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L12-R14) [[12]](diffhunk://#diff-f5518895f75f179e6bcfaa190afe01fa4b96d63e6cd781d7491e17265512d759L282-R267)

**Testing:**

* Added unit tests for both `scrub_description` and `strip_comments_from_prompt` to verify correct handling of edge cases, truncation, HTML stripping, and comment removal. [[1]](diffhunk://#diff-e7f4a911f10623ee8e02d1fa8597736ecdf002275d9e054765f3ac25b4a2ea8cR1-R27) [[2]](diffhunk://#diff-8de5f1c6252e75852a9faf8ec18345d41845c3017450b74fac26475ba12e4f8cR1-R44)

These changes improve the reliability and clarity of prompts sent to the LLM, reduce accidental prompt injection or hallucination triggers, and centralize text normalization logic for easier maintenance.

## Test plan

- [x] `PYTHONPATH=src pytest tests/` passes
- [x] `ruff check src/ tests/` passes
- [ ] Frontend (not touched)
- [x] CHANGELOG.md updated

### CODE REVIEW

File: benchmarks/llm/src/benchmark/runner.py [hunk 1]
```diff
@@ -177,7 +178,7 @@ def _build_user_prompt(
     addressing_mode: str = "timestamps",
     id_segments: list[dict] | None = None,
 ) -> str:
-    description = (episode.metadata.description or "").strip()
+    description = scrub_description(episode.metadata.description, max_length=800)
```
Comment (mrfootoyou) [inline benchmarks/llm/src/benchmark/runner.py]: 800 chars seems reasonable. Many/most episode descriptions do not contain useful context toward the end.

File: src/ad_detector/__init__.py [hunk 2]
```diff
@@ -1596,12 +1601,15 @@ def detect_ads(self, segments: list[dict], podcast_name: str = "Unknown",
 
             # Prepare description section (shared across windows)
             description_section = ""
+            podcast_description = scrub_description(podcast_description, max_length=200)
```
Comment (mrfootoyou) [inline src/ad_detector/__init__.py]: 200 chars seems reasonable for podcast description. This text is unlikely to add much context anyway.

File: benchmarks/llm/src/benchmark/runner.py (hunk 1 above)
Comment (ttlequals0) [reply benchmarks/llm/src/benchmark/runner.py]: 800 seems a bit too low. I spot-checked some of my episodes, and most passed that; also, the useful part toward the end is usually the "sponsored by x, y, and z" section.

File: src/ad_detector/__init__.py (hunk 2 above)
Comment (ttlequals0) [reply src/ad_detector/__init__.py]: This is also a bit too short; I have a few that exceed this and get truncated in odd places at 200. I'm pulling data from my instance to see the min, avg, and max across episodes and podcasts.

**Comment by ttlequals0** [comment]
Real production data (32 feeds, 1,729 episodes), measuring the post-scrub length that the cap actually truncates:

| Scope        | mean | median | p90  | p95  | p99  | max   |
|--------------|------|--------|------|------|------|-------|
| Podcast desc | 301  | 304    | 511  | 583  | 644  | 668   |
| Episode desc | 1353 | 946    | 2530 | 3540 | 9751 | 89160 |


Coverage at suggested caps: episode description 800 -> only 44% fit; 4000 -> 96.7%; podcast description 200 -> 34%; 800 -> 100%.
